### PR TITLE
Add IT tests for PushDownEnrich rule

### PR DESCRIPTION
**Pull request by dnhatn**
This PR adds some queries that could fail without the PushDownEnrich rule.

### CODE REVIEW

**Review by luigidellaquila — COMMENTED** [review]
Thanks Nhat

#### tl;dr: 
This is not a simple optimization rule, it also serves as a way to avoid unbounded sorts.
Removing this rule will break some queries (though not so relevant probably).

On the other hand, the rule seems to be too optimistic on pushing down without checking if it's really possible, so it will need a double-check for sure.

I don't know which is the biggest problem TBH, so I'm ok with deleting it if the team agrees.

#### Longer explanation below:

<details>
  <summary>Click here</summary>

This not a pure optimization rule, its main purpose is described in the PushDownEval javadoc:
```
Although it seems arbitrary whether the OrderBy or the Eval is executed first,
this transformation ensures that OrderBys only separated by an eval 
can be combined by PushDownAndCombineOrderBy
```

A query like

```
from idx
| eval x = foo + bar 
| sort x 
| enrich ... 
| sort foo
```

will become problematic.
With current logic, the two sorts will be merged and then eventually become a TopN; removing this rule, the first SORT won't have a LIMIT (and in this case it won't be possible to push it down to Lucene either, because it's on a calculated value), so the query planning would fail with 

```
"type": "esql_illegal_argument_exception",
"reason": "unknown physical plan node [OrderExec]",
```

About Project, I think the original reasoning is the same: if you have `SORT | PROJECT | EVAL | SORT` you want to have `SORT | EVAL | PROJECT | SORT` and  then `EVAL | SORT | PROJECT | SORT`, because then `PushDownAndCombineOrderBy` will [take care](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java#L840) of pushing SORT past PROJECT


To be clear, this is how the problem of unbounded sort was addressed since the beginning, but I'm not sure it's the best solution. Probably there are other ways to detect and prune redundant sorts.
Moreover, we won't be able to avoid unbounded sorts in all cases (eg. for some combinations with MV_EXPAND), so we could decide to live with it.

And btw, these rules have a problem: you cannot push EVAL or ENRICH past SORT if the sort is on a column that is calculated by ENRICH/EVAL. We should at least add a check for this.

As a conclusion, it's OK for me to remove it, as the impacted use cases are very limited and probably irrelevant in terms of functionality (ie. users can easily remove a redundant SORT if we help them identify it with proper error messages).

</summary>

**Comment by dnhatn** [comment]
@luigidellaquila, thank you for your feedback. I see the issue with the Order, and I'll include that test in this pull request. Regarding the project rule, it appears there's no issue without it, correct?

**Comment by luigidellaquila** [comment]
The rule is not problematic for Project alone, but it could make a difference for SORT + PROJECT.
Eg. if you disable the second part of the rule (the pushdown past Project), the following query will fail

```
from person | eval x = sequence+1 | sort x | keep tag, x, sequence | enrich test-policy on tag with anumber | sort sequence | keep tag, anumber
```

**Comment by dnhatn** [comment]
Thanks @luigidellaquila. These cases are really helpful. I will update this PR to add these tests + comment and leave the rule as is.

**Comment by dnhatn** [comment]
Costin pointed out that the optimized physical plan should handle this. I've updated this PR with the tests suggested by Luigi.

**Comment by costin** [comment]
For the record, the extraction is handled in the physical plan - the logical plan goal is to simplify and normalize the plan, as much as possible.
The following query:
```
var plan = plan("from test | keep first_name | enrich foo on first_name");
```
inside LocalPhysicalPlanOptimizerTest yields the following 
logical plan:
```
EsqlProject[[first_name{f}#5, a{r}#16, b{r}#17]]
\_Enrich[ANY,[66 6f 6f][KEYWORD],first_name{f}#5,{"match":{"indices":[],"match_field":"fld","enrich_fields":["a","b"]}},{=i
dx},[a{r}#16, b{r}#17]]
  \_Limit[500[INTEGER]]
    \_EsRelation[test][_meta_field{f}#10, emp_no{f}#4, first_name{f}#5, ge..]
```
physical plan:
```
ProjectExec[[first_name{f}#5, a{r}#16, b{r}#17]]
\_EnrichExec[ANY,first_name{f}#5,foo,fld,{=idx},[a{r}#16, b{r}#17]]
  \_LimitExec[500[INTEGER]]
    \_ExchangeExec[[],false]
      \_ProjectExec[[first_name{f}#5]]
        \_FieldExtractExec[first_name{f}#5][]
          \_EsQueryExec[test], query[][_doc{f}#18], limit[500], sort[] estimatedRowSize[66]
```
Notice the field extraction only occurs on the necessary field, not everything else.

@dnhatn thanks for raising this, let's get the IT tests in for consistency - no need to linger anymore on this PR.
@luigidellaquila I've assigned #105043 to you.

**Review by costin — APPROVED** [review]
LGTM

**Comment by dnhatn** [comment]
@luigidellaquila @costin Thanks for the feedback.
